dali/operators/decoder/audio/utils.h

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_UTILS_H_
+#define DALI_OPERATORS_DECODER_AUDIO_UTILS_H_
+
+#include <functional>
+#include <numeric>
+#include <utility>
+#include <vector>
+#include <cassert>
+#include <cmath>
+#include "dali/core/error_handling.h"
+#include "dali/core/span.h"
+
+namespace dali {
+
+/**
+ * Downmixing data to mono-channel. It's safe to do it in-place.
```

Comment:
```suggestion
 * Downmixing data to mono-channel. It's safe to do it in-place.
```
It's NOT safe to do it in place. It interferes with loop unrolling and automatic vectorization.